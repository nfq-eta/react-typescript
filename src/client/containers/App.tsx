import * as React from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import CheckBox, { IItem } from '../components/CheckBox';
import * as CheckboxItemsJson from '../components/tests/__mocks__/CheckboxItems.json';
import * as CheckboxSelectedItemsJson from '../components/tests/__mocks__/CheckboxSelectedItems.json';
import { addCheckbox } from '../modules/checkbox/actions';
import { IRootState } from '../core/reducers';

export interface IAppProps {
    actions: typeof addCheckbox;
    items: IItem[];
}

export interface IAppState {
    items: IItem[];
}

// @connect(mapStateToProps, mapDispatchToProps)
class App extends React.Component<IAppProps, IAppState> {
    selectedItems = new Map(CheckboxSelectedItemsJson);

    constructor() {
        super();

        this.state = { items: CheckboxItemsJson };
    }

    handleClick = (item: IItem, checked: boolean) => {
        if (checked) {
            this.selectedItems.set(item.value, item);
        } else {
            this.selectedItems.delete(item.value);
        }
    };

    isSelected(item: IItem): boolean {
        return this.selectedItems.has(item.value);
    }

    handleAdd = () => {
        this.props.actions({
            value: 'demo',
            label: 'Demo',
        });
    };

    render() {
        return (
            <div>
                <h1>App</h1>
                <button onClick={this.handleAdd}>Add more</button>
                {this.props.items.map(item => (
                    <CheckBox
                        item={item}
                        key={'a' + item.value}
                        checked={this.isSelected(item) || false}
                        handleClick={this.handleClick}
                    />
                    ))
                }
            </div>
        );
    }
}

function mapStateToProps(state: IRootState) {
    return ({
        items: state.checkBoxItems,
    });
}

function mapDispatchToProps(dispatch: any) {
    return ({
        actions: bindActionCreators(addCheckbox as any, dispatch),
    });
}

export default connect(mapStateToProps, mapDispatchToProps)(App);

export { App as AppDisconnected };
