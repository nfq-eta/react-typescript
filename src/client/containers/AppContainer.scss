.button {
  color: #0700ff;
}
