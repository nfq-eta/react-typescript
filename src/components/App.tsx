import * as React from 'react';
import CheckBox, { IItem } from './CheckBox';

import CheckboxItemsJson = require('./tests/__mocks__/CheckboxItems.json');
import CheckboxSelectedItemsJson = require('./tests/__mocks__/CheckboxSelectedItems.json');

export interface IAppProps {
    title?: string;
}

export interface IAppState {
    items: IItem[];
}

export default class App extends React.Component<IAppProps, IAppState> {
    selectedItems = new Map(CheckboxSelectedItemsJson);

    constructor() {
        super();

        this.state = { items: CheckboxItemsJson };
    }

    handleClick = (item: IItem, checked: boolean) => {
        if (checked) {
            this.selectedItems.set(item.id, item);
        } else {
            this.selectedItems.delete(item.id);
        }
    };

    isSelected(item: IItem): boolean {
        return this.selectedItems.has(item.id);
    }

    render() {
        return (
            <div>
                <h1>{this.props.title || ''}</h1>
                {this.state.items.map(item => (
                    <CheckBox
                        item={item}
                        key={item.id}
                        checked={this.isSelected(item) || false}
                        handleClick={this.handleClick}
                    />
                    ))
                }
            </div>
        );
    }
}
