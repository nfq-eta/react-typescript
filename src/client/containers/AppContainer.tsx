import * as React from 'react';
import { view } from 'react-easy-state';

import { CheckBox, IItem } from '../components/checkbox/CheckBoxComponent';
import { store } from '../core/store';

import * as styles from './AppContainer.scss';

export type AppProps = {
  items?: IItem[];
};

export class App extends React.Component<AppProps, {}> {
  static defaultProps = {
    items: [],
  };

  handleAdd = () => {
    store.foo = store.foo == 'clicked' ? 'click me' : 'clicked';
    store.addAction({
      id: Math.round(new Date().getTime() / 1000).toString(),
      label: 'Demo',
      value: 'demo',
    });
  };

  render() {
    return (
      <div>
        <button onClick={this.handleAdd} className={styles.button}>
          Add more
        </button>
        {store.checkBoxItems.map((item, index) => (
          <div key={index}>
            <CheckBox item={item} />
            <button>Delete</button>
          </div>
        ))}
      </div>
    );
  }
}

export default view(App);
