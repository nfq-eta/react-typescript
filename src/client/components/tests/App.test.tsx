import * as React from 'react';
import { shallow } from 'enzyme';
import enzymeToJson from 'enzyme-to-json';
import {} from 'jest';

import { AppDisconnected as App } from '../../containers/App';

describe('App.tsx', () => {
    const props = {
        actions: jest.fn(),
        items: [],
    };

    it('renders without crashing', () => {
        shallow(<App {...props} />);
    });

    it('matches snapshot', () => {
        const wrapper = shallow(<App {...props} />);
        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });

    it('Title should be visible', () => {
        const wrapper = shallow(<App {...props} />);
        expect(wrapper.find('h1').text()).toContain('App');
    });

    it('should handle click', () => {
        const wrapper = shallow(<App {...props} />);
        const instance = wrapper.instance() as App;
        const params = { value: 1, label: 'Demo' };

        instance.handleClick(params, true);
        expect(instance.selectedItems.has(1)).toBeTruthy();

        instance.handleClick(params, false);
        expect(instance.selectedItems.has(1)).toBeFalsy();
    });
});
