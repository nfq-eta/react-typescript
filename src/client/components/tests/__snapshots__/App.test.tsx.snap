// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.tsx matches snapshot 1`] = `
<div>
  <h1>
    App
  </h1>
  <button
    onClick={[Function]}
  >
    Add more
  </button>
</div>
`;
